saml_provider/app.py

```diff
@@ -0,0 +1,81 @@
+import boto3
+import json
+import logging
+import requests
+
+from crhelper import CfnResource
+
+logger = logging.getLogger(__name__)
+helper = CfnResource(
+    json_logging=False, log_level='DEBUG', boto_level='CRITICAL')
+
+try:
+    client = boto3.client("iam")
+except Exception as e:
+    helper.init_failure(e)
+
+def get_parameters(event):
+    name = event['ResourceProperties']['Name']
+    url = event['ResourceProperties']['URL']
+    if 'Metadata' in event['ResourceProperties']:
+        metadata = event['ResourceProperties']['Metadata']
+    else:
+        metadata = get_metadata(url)
+
+    return name, metadata, url
+
+def get_metadata(url):
+    response = requests.get(url)
+    metadata = response.text
+    logger.info("metadata = " + metadata)
+    if response.status_code == 200:
```

Comment:
Is there a situation where you'd want to continue w/o metadata if not a 200? If not probably better to raise an exception here for another satus code rather than let it continue to create_provider with no meta data.

Reply:
fixed